src/overlays/actors/ovl_En_Dekunuts/z_en_dekunuts.c

```diff
@@ -71,85 +121,634 @@ static DamageTable D_808BEF64 = {
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BDFB8.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE1CC.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE22C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE294.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE358.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE3A8.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE3FC.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE484.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE4D4.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE680.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE6C4.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BE73C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/EnDekunuts_Update.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BEBD0.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/func_808BED30.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dekunuts_0x808BD1E0/EnDekunuts_Draw.asm")
+void EnDekunuts_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDekunuts* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 35.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06002468, &D_0600326C, this->jointTable, this->morphTable, 10);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->unk_194 = (this->actor.params >> 8) & 0xFF;
+    thisx->params &= 0xFF;
+    if ((this->unk_194 == 0xFF) || (this->unk_194 == 0)) {
+        this->unk_194 = 1;
+    }
+
+    if (this->actor.params == 1) {
+        this->actor.flags &= -2;
+        this->collider.base.colType = COLTYPE_NONE;
+        this->collider.info.bumperFlags |= 0x78;
+    } else if (this->actor.params == 2) {
+        this->actor.targetMode = 0;
+    }
+
+    func_808BD428(this);
+}
+
+void EnDekunuts_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDekunuts* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_808BD348(EnDekunuts* this) {
+    this->unk_18E = 10;
+    this->unk_214 = 0.55f;
+    this->unk_218 = 0.82500005f;
+    this->unk_210 = 1.0f;
+    this->collider.base.colType = COLTYPE_HIT3;
+    this->unk_190 = 0x50;
```

Comment:
```suggestion
    this->unk_190 = 80;
```
This seems to be a timer